coding/python_style_guide.rst

```diff
@@ -12,648 +13,671 @@ To request changes to these standards, please file an :ref:`RFC <decision-making
-- ``I`` (uppercase letter eye) as single character variable names.
+Line Length MUST be less than or equal to 110 columns
+-----------------------------------------------------
 
-In some fonts, these characters are indistinguishable from the numerals one and zero.
-When tempted to use ``l``, use ``L`` instead.
+Limit all lines to a maximum of 110 characters.
+This conforms to the :doc:`cpp_style_guide` (see :ref:`4-6 <style-guide-cpp-4-6>`).
 
-.. _style-guide-py-inheritance:
+This differs from the `PEP 8 recommendation of 79 characters <https://www.python.org/dev/peps/pep-0008/#id19>`_.
 
-3. Designing for Inheritance
-============================
+.. _style-guide-py-implied-continuation:
 
-Always decide whether a class's methods and instance variables (collectively: "attributes") should be public or non-public.
-If in doubt, choose non-public; it's easier to make it public later than to make a public attribute non-public.
+Python's implied continuation inside parens, brackets and braces SHOULD be used for wrapped lines
+-------------------------------------------------------------------------------------------------
 
-Public attributes are those that you expect unrelated clients of your class to use, with your commitment to avoid backward incompatible changes.
-Non-public attributes are those that are not intended to be used by third parties; you make no guarantees that non-public attributes won't change or even be removed.
+The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces.
+If necessary, you can add an extra pair of parentheses around an expression, but sometimes using a backslash looks better.
+Make sure to indent the continued line appropriately. Some examples:
 
-We don't use the term "private" here, since no attribute is really private in Python (without a generally unnecessary amount of work).
-Another category of attributes are those that are part of the "subclass API" (often called "protected" in other languages).
-Some classes are designed to be inherited from, either to extend or modify aspects of the class's behavior.
-When designing such a class, take care to make explicit decisions about which attributes are public, which are part of the subclass API, and which are truly only to be used by your base class.
+.. code-block:: py
 
-For simple public data attributes, it is best to expose just the attribute name, without complicated accessor/mutator methods.
-Keep in mind that Python provides an easy path to future enhancement, should you find that a simple data attribute needs to grow functional behavior.
-In that case, use properties to hide functional implementation behind simple data attribute access syntax.
+    class Rectangle(Blob):
+        """Documentation for Rectangle.
+        """
+        def __init__(self, width, height,
+                     color='black', emphasis=None, highlight=0):
+            if width == 0 and height == 0 and
```

Comment:
This example code is syntactically broken so I'm not sure what to make of it. It's broken because it doesn't do continuation properly but I think it's point should be that the first `if` statement uses `\` for continuation (those are missing) and the second one uses parens. It might be worth noting that:
```python
if (width == 0
    height == 0):
    do_something()
```
can be visually confusing with the last line of the `if` lining up with the statement inside the block.